docs/maven-tutorial.md

```diff
@@ -0,0 +1,222 @@
+
+Download the official Nexus Docker image and use `new-app` to deploy it.
+```bash
+docker pull sonatype/nexus
+oc new-app sonatype/nexus
+```
+
+After the build has finished, run:
+```bash
+oc get svc
+```
+
+The output should resemble:
+```bash
+NAME      CLUSTER-IP       EXTERNAL-IP   PORT(S)    AGE
+nexus     172.30.170.249   <none>        8081/TCP   6s
+```
+
+Confirm that Nexus is running by navigating your browser to the new Cluster-IP
+at port 8081.  The server may take a bit of time to start, be patient!
+
+> NOTE: Nexus comes pre-configured for the Central Repository, but you may need others
+> for your application.  At this point, add any special repositories to Nexus.
+
+Now we'll create a Persistent Volume for Nexus, so that our dependencies will
+not be lost when the Pod running the server terminates.  If you do not have admin access on your cluster, ask
+your system administrator to create a Read/Write Persistent Volume for you, and
+skip to the section on creating Persistent Volume Claim.
+
+> NOTE: If you don't want persistent storage, continue to
+> [Modifying the Builder](#modifying-the-builder). However, your cached
+> dependencies will be lost with the pod.
+
+Create a directory on your host machine that will be writable by OpenShift AND
+persistent, in case your host machine reboots.
+```bash
+mkdir /nexus-pv
+chmod 777 /nexus-pv
+```
+
+Log in to your admin user, if necessary. Using `oc cluster up`, this is done
+with:
+```bash
+oc login -u system:admin
+```
+
+Create a file called `pv.yaml` with the following content:
+```yaml
+---
+apiVersion: v1
+kind: PersistentVolume
+metadata:
+  name: nexus-pv
+  labels:
+    type: local
+spec:
+  hostPath:
+    path: "/nexus-pv"
+  accessModes:
+  - ReadWriteOnce
+  capacity:
+    storage: 1Gi
+  persistentVolumeReclaimPolicy: Retain
+```
+
+Import that file into OpenShift.
+```bash
+oc create -f pv.yaml
+```
+
+Log back in to your user account and add a Persistent Volume Claim to the Nexus
+Deployment Config.  Using `oc cluster up`, the default user is 'developer'
+```bash
+oc login -u developer
+oc volumes dc/nexus --remove --confirm
+oc volumes dc/nexus --add \
+	--name 'nexus-pvc' \
+	--type 'pvc' \
+	--mount-path '/sonatype-work/' \
+	--claim-name 'nexus-pv' \
+	--claim-size '1G'
+```
+
+This removes the previous volume for the Deployment Config, and adds a claim
+for 1 gigabyte of storage mounted at `/sonatype-work`, which is where our
+dependencies will be stored.  Due to the change in config, the Nexus image will
+be redeployed automatically.
+
+Verify that Nexus is running by refreshing the Nexus page in your browser. You
+can monitor the builds progress easily using
+```bash
+oc get pods -w
+```
+
+## Modifying the Builder
+
+The next steps will be modifying our builder image to use this repository.
+The rest of the tutorial will be editing
+[this sample repository](https://github.com/bparees/openshift-jee-sample.git),
+but these changes should work for any project.
```

Comment:
I'd like to show how to do this without using the environment variable, since not every image may implement it.